app/gui2/src/components/widgets/NumericInputWidget.vue

```diff
@@ -21,6 +22,16 @@ const MIN_CONTENT_WIDTH = 56
 
 // Edited value reflects the `modelValue`, but does not update it until the user defocuses the field.
 const editedValue = ref('')
+// Last value which is a parseable number. It's a string, because the Enso number literals differ from js
+// representations.
+const lastValidValue = ref<string>()
+watch(editedValue, (newValue, oldValue) => {
+  if (newValue != oldValue) {
+    if (newValue == '' || is_numeric_literal(newValue)) {
```

Comment:
I think it would be nice to re-export `is_numeric_literal` from a higher level module with our JS naming convention. Whether it is an FFI call is an implementation detail.